web-admin/src/features/projects/environment-variables/EditDialog.svelte

```diff
@@ -0,0 +1,354 @@
+  import { page } from "$app/stores";
+  import { Button } from "@rilldata/web-common/components/button";
+  import {
+    Dialog,
+    DialogContent,
+    DialogHeader,
+    DialogTitle,
+    DialogTrigger,
+    DialogDescription,
+    DialogFooter,
+  } from "@rilldata/web-common/components/dialog-v2";
+  import Checkbox from "@rilldata/web-common/components/forms/Checkbox.svelte";
+  import {
+    createAdminServiceUpdateProjectVariables,
+    getAdminServiceGetProjectVariablesQueryKey,
+  } from "@rilldata/web-admin/client";
+  import { type AdminServiceUpdateProjectVariablesBodyVariables } from "@rilldata/web-admin/client";
+  import { useQueryClient } from "@tanstack/svelte-query";
+  import { eventBus } from "@rilldata/web-common/lib/event-bus/event-bus";
+  import { defaults, superForm } from "sveltekit-superforms";
+  import { yup } from "sveltekit-superforms/adapters";
+  import { object, string } from "yup";
+  import { EnvironmentType } from "./types";
+  import Input from "@rilldata/web-common/components/forms/Input.svelte";
+  import { onMount } from "svelte";
+
+  export let open = false;
+  export let id: string;
+  export let environment: string;
+  export let name: string;
+  export let value: string;
+  export let variableNames: string[] = [];
+
+  let initialEnvironment: {
+    isDevelopment: boolean;
+    isProduction: boolean;
+  };
+  let isDevelopment: boolean;
+  let isProduction: boolean;
+  let isKeyAlreadyExists = false;
+  let inputErrors: { [key: number]: boolean } = {};
+
+  $: organization = $page.params.organization;
+  $: project = $page.params.project;
+
+  $: isEnvironmentSelected = isDevelopment || isProduction;
+  $: hasChanges =
+    $form.key !== initialValues.key ||
+    $form.value !== initialValues.value ||
+    initialEnvironment?.isDevelopment !== isDevelopment ||
+    initialEnvironment?.isProduction !== isProduction;
+  $: hasExistingKeys = Object.values(inputErrors).some((error) => error);
+
+  const queryClient = useQueryClient();
+  const updateProjectVariables = createAdminServiceUpdateProjectVariables();
+
+  const initialValues = {
+    environment: environment,
+    key: name,
+    value: value,
+  };
+
+  const schema = yup(
+    object({
+      environment: string().optional(),
+      key: string()
+        .optional()
+        .matches(
+          /^[a-zA-Z0-9_]+$/,
+          "Key must only contain letters, numbers, and underscores",
+        ),
+      value: string().optional(),
+    }),
+  );
+
+  const {
+    form,
+    enhance,
+    formId,
+    submit,
+    errors,
+    allErrors,
+    submitting,
+    reset,
+  } = superForm(defaults(initialValues, schema), {
+    // See: https://superforms.rocks/concepts/multiple-forms#setting-id-on-the-client
+    id: id,
+    SPA: true,
+    validators: schema,
+    async onUpdate({ form }) {
+      if (!form.valid) return;
+      const values = form.data;
+
+      const flatVariables = {
+        [values.key]: values.value,
+      };
```

Comment:
Nit: should this be the singular `flatVariable`?

Reply:
right, singular, updating it